hwy/contrib/unroller/unroller_test.cc

```diff
@@ -0,0 +1,295 @@
+// Copyright Google LLC 2021
+//           Matthew Kolbe 2023
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <limits>;
+#include <iostream>;
+
+// clang-format off
+#undef HWY_TARGET_INCLUDE
+#define HWY_TARGET_INCLUDE "hwy/contrib/unroller/unroller_test.cc"
+#include "hwy/foreach_target.h"  // IWYU pragma: keep
+#include "hwy/highway.h"
+#include "hwy/contrib/unroller/unroller-inl.h"
+#include "hwy/tests/test_util-inl.h"
+// clang-format on
+
+#define UNUSED(x) (void)(x)
+
+HWY_BEFORE_NAMESPACE();
+namespace hwy {
+namespace HWY_NAMESPACE {
+
+template <typename T>
+T SimpleDot(const T* pa, const T* pb, size_t num) {
+  T sum = 0;
+  for (size_t i = 0; i < num; ++i) {
+    sum += pa[i] * pb[i];
+  }
+  return sum;
+}
+
+template <typename T>
+T SimpleAcc(const T* pa, size_t num) {
+  T sum = 0;
+  for (size_t i = 0; i < num; ++i) {
+    sum += pa[i];
+  }
+  return sum;
+}
+
+template <typename T>
+T SimpleMin(const T* pa, size_t num) {
+  T min = std::numeric_limits<T>::max();
+  for (size_t i = 0; i < num; ++i) {
+    if(min > pa[i])
+        min = pa[i];
+  }
+  return min;
+}
+
+template <typename T>
+struct accv : UnrollerUnit<accv<T>, T, T> {
+  using TT = hn::ScalableTag<T>;
+  inline hn::Vec<TT> Func(ptrdiff_t const& HWY_RESTRICT idx,
+                          hn::Vec<TT> const& HWY_RESTRICT x,
+                          hn::Vec<TT> const& HWY_RESTRICT y) const {
+    return x + y;
+  }
+
+  inline ptrdiff_t StoreImpl(ptrdiff_t const& HWY_RESTRICT idx,
+                             T* HWY_RESTRICT to, const hn::Vec<TT> x) const {
+    // no stores in a reducer
+    return 0;
+  }
+
+  inline ptrdiff_t MaskStoreImpl(const ptrdiff_t idx, T* HWY_RESTRICT to,
+                                 hn::Vec<TT> const& HWY_RESTRICT x,
+                                 ptrdiff_t const& HWY_RESTRICT places) const {
+    // no stores in a reducer
+    return 0;
+  }
+
+  inline ptrdiff_t ReduceImpl(hn::Vec<TT> const& HWY_RESTRICT x,
```

Comment:
Even Vec can be passed around by value (on MSVC that requires /Gv to work well). Then we can/should also remove the HWY_RESTRICT for vectors.